src/cascadia/TerminalConnection/AzureConnection.cpp

```diff
@@ -0,0 +1,885 @@
+            catch (...)
+            {
+                if (s != "n" && s != "r")
+                {
+                    _outputHandlers(winrt::to_hstring(invalidAccessInput));
+                    return;
+                }
+                else if (s == "r")
+                {
+                    std::lock_guard<std::mutex> lg{ _commonMutex };
+                    _removeOrNew = true;
+                    _canProceed.notify_one();
+                    return;
+                }
+                else
+                {
+                    std::lock_guard<std::mutex> lg{ _commonMutex };
+                    _removeOrNew = false;
+                    _canProceed.notify_one();
+                    return;
+                }
+            }
+            if (storeNum >= _maxStored)
+            {
+                _outputHandlers(winrt::to_hstring(numOutOfBoundsError));
+                return;
+            }
+            std::lock_guard<std::mutex> lg{ _commonMutex };
+            _storedNumber = storeNum;
+            _canProceed.notify_one();
+            return;
+        }
+        // The user has multiple tenants in their Azure account, let them choose one of them
+        case State::tenantChoice:
+        {
+            int tenantNum = -1;
+            try
+            {
+                tenantNum = std::stoi(winrt::to_string(data));
+            }
+            catch (...)
+            {
+                _outputHandlers(winrt::to_hstring(nonNumberError));
+                return;
+            }
+            if (tenantNum >= _maxSize)
+            {
+                _outputHandlers(winrt::to_hstring(numOutOfBoundsError));
+                return;
+            }
+            std::lock_guard<std::mutex> lg{ _commonMutex };
+            _tenantNumber = tenantNum;
+            _canProceed.notify_one();
+            return;
+        }
+        // User has the option to save their connection settings for future logins
+        case State::storeTokens:
+        {
+            std::string s = winrt::to_string(data);
+            if (s != "y" && s != "n")
+            {
+                _outputHandlers(winrt::to_hstring(invalidStoreInput));
+                return;
+            }
+            else if (s == "y")
+            {
+                std::lock_guard<std::mutex> lg{ _commonMutex };
+                _store = true;
+                _canProceed.notify_one();
+            }
+            else
+            {
+                std::lock_guard<std::mutex> lg{ _commonMutex };
+                _store = false;
+                _canProceed.notify_one();
+            }
+            return;
+        }
+        // We are connected, send user's input over the websocket
+        case State::termConnected:
+        {
+            websocket_outgoing_message msg;
+            const auto str = winrt::to_string(data);
+            msg.set_utf8_message(str);
+
+            auto msgSentTask = _cloudShellSocket.send(msg);
+            msgSentTask.wait();

```

Comment:
Out of curiousity, do we want to wait here? Won't that block our input on the websocket request? What happens if we just send the websocket request, then immediately return to process more input. Will the websocket be smart enough to be able to keep the order of the messages on the other side?

Reply:
Honestly, I'm not sure. I just figured that if we don't wait then there might be situations where there's more than one message trying to be sent at the same time and then the order of them being sent (from our side) may get jumbled up - and then I'm not sure if/how the other side can tell the correct order. If we're sure that won't happen I can go ahead and remove the wait

Reply:
Removed the wait in the latest version!